// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.pair.job;

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

// The PAIR job type.
// Next ID: 3
enum JobType {
  JOB_TYPE_UNSPECIFIED = 0;
  // Publisher Pair List Job
  JOB_TYPE_GENERATE_PUB_PAIR_LIST = 1;
  // Advertiser/Publisher Match Job
  JOB_TYPE_MATCH = 2;
}

// Next ID: 3
message AttestationInfo {
  string project_id = 1;
  string wip_provider = 2;
}

// The PAIR job data.
// Next ID: 12
message PairJobData {
  JobType job_type = 1;
  string publisher_input_bucket = 2;
  // Only used for Publisher list generation.
  string publisher_metadata_blob_path = 3;
  // Only used for Publisher list generation.
  string publisher_user_list_blob_path = 4;
  // Only used for matching.
  string publisher_mapping_blob_path = 5;
  // Only used for matching.
  string advertiser_input_bucket = 6;
  // Only used for matching.
  string advertiser_user_list_blob_path = 7;
  // Only used for matching.
  string match_output_bucket = 8;
  // Only used for matching.
  string match_list_blob_path = 9;

  optional AttestationInfo publisher_bucket_attestation_info = 10;
  // Only used for matching.
  optional AttestationInfo advertiser_bucket_attestation_info = 11;
}
